sql/index.ts

```diff
@@ -132,12 +132,123 @@ export class Sql {
   }
 
   public asString(castableToString: string) {
-    if (this.dialect === "postgres") {
+    if (this.dialect === "postgres" || this.dialect === "redshift") {
       return `cast(${castableToString} as varchar)`;
     }
     return `cast(${castableToString} as string)`;
   }
 
+  // Surrogate keys.
+
+  public surrogateKey(columnNames: string[]) {
+    const columnsAsStrings = columnNames.map(id => this.asString(id)).join(`,`);
+    if (this.dialect === "standard") {
+      return this.asString(`farm_fingerprint(concat(${columnsAsStrings}))`);
+    }
+    if (
```

Comment:
you could just leave this as the default, so special case standard, mssql, then afterwards just return this expression and remove the if.